hicommonwealth/commonwealth
# Apply weighted voting to reactions

## Link to Issue
Closes: #9133
Closes: #9141
Closes: #9139 
Closes: #9261

## Description of Changes
- Applies topic-based weighted voting to thread reactions
- Updates the createTopic command to set weighted voting options
- Updates CreateContestManagerMetadata command to prevent creation of recurring contest if stake not enabled
- Adds migration that makes all non-general topics in a staked community to be staked

## Test Plan
- Go to a non-staked community
  - Create topic + thread + reaction – ensure no errors
- Enable stake in the community + buy stake + ensure you have > 0 stake
  - Create a thread in General, vote on it– ensure that the reaction weight is 1
  - Create a thread in a non-General topic, vote on it– ensure that the reaction weight is based on stake
  - Create a new topic– ensure no errors
  - Update a topic– ensure no errors

## Deployment Plan
N/A

## Other Considerations
- All new topics created in a staked community will be staked by default since the UI is not wired up. Once the UI is wired up, then the stake params from the UI will be used (stake/ERC20/none)– that code path is activated via the `FLAG_FARCASTER_CONTEST` env var
